### PR TITLE
Feature: add step 5.1

### PR DESCRIPTION
**Pull request by jonathanrocher**
New step that takes 4.1 as a starting point and adds the ability to view an image folder in the application:
Screen Shot 2022-05-29 at 8 12 31 PM

To do this, all files are identical except the addition of:
- `stage5.1_fuller_application/pycasa/model/image_folder.py`
- `stage5.1_fuller_application/pycasa/model/tests/test_image_folder.py`
- `stage5.1_fuller_application/pycasa/ui/image_folder_editor.py`
- `stage5.1_fuller_application/pycasa/ui/image_folder_view.py`

and the update of the task file to support folders:
- `stage5.1_fuller_application/pycasa/ui/tasks/pycasa_task.py`

Note: this version doesn't have the ability to count the number of faces.

### CODE REVIEW

**Comment by jonathanrocher** [comment]
I didn't realize @siddhantwahal created very similar models and views for a folder, so I will update this PR once #31 is merged.

File: stage5.1_fuller_application/pycasa/model/image_folder.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    data = Instance(pd.DataFrame)
+
+    def __init__(self, **traits):
+        # Don't forget this!
+        super(ImageFolder, self).__init__(**traits)
+        if not os.path.isdir(self.path):
+            msg = f"Unable to create an ImageFolder from {self.path} since" \
+                  f" it is not a valid directory."
+            raise ValueError(msg)
```
Comment (siddhantwahal) [inline stage5.1_fuller_application/pycasa/model/image_folder.py:27]: Can this also be accomplished with `path = Directory(exists=True)`?

Comment (jonathanrocher) [reply stage5.1_fuller_application/pycasa/model/image_folder.py:27]: That's true, but 1. I didn't know about it and 2. my goal was to introduce the `__init__` thing which is more fundamental. Maybe in the training, we can have a section about exploring Trait's keyword args, and show that off with this. Thanks @siddhantwahal !

Comment (jonathanrocher) [reply stage5.1_fuller_application/pycasa/model/image_folder.py:27]: Ps First step is me replacing my models with yours anyway...

Comment (siddhantwahal) [reply stage5.1_fuller_application/pycasa/model/image_folder.py:27]: > That's true, but 1. I didn't know about it and 2. my goal was to introduce the `__init__` thing which is more fundamental. Maybe in the training, we can have a section about exploring Trait's keyword args, and show that off with this and maybe one Editor keyword arg. In fact, I already have a slide for this! Thanks @siddhantwahal !

Sounds good! I actually didn't know about this, but there's a `trait_init` method which automatically calls `super().__init__`: https://docs.enthought.com/traits/traits_tutorial/introduction.html?highlight=initializer#the-traits-init-method

**Comment by jonathanrocher** [comment]
Planning on merging this soon-ish so I can push the next PR and we can continue to build up...

**Review by siddhantwahal — APPROVED** [review]
Looks great!

**Comment by jonathanrocher** [comment]
Thank you @siddhantwahal !

**Pull request merged**